pkgs/development/ocaml-modules/fmt/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchurl, ocaml, findlib, ocamlbuild, opam, topkg, cmdliner }:
+
+stdenv.mkDerivation {
+  name = "ocaml${ocaml.version}-fmt-0.8.0";
+
+  src = fetchurl {
+    url = http://erratique.ch/software/fmt/releases/fmt-0.8.0.tbz;
+    sha256 = "16y7ibndnairb53j8a6qgipyqwjxncn4pl9jiw5bxjfjm59108px";
+  };
+
+  unpackCmd = "tar xjf $src";
+
+  buildInputs = [ ocaml findlib ocamlbuild opam topkg cmdliner ];
+
+  inherit (topkg) buildPhase;
+
+  createFindlibDestdir = true;
+
+  installPhase = ''
```

Comment:
What do you mean by “cool”? Should derivations be cool? I don’t see how this `installPhase` value could be shared among several derivations as it explicitly mentions the library name (twice). It is true however that the common and recurring parts could be factorized away in a library for building OCaml derivations.